components/layout_2020/geom.rs

```diff
@@ -364,79 +300,45 @@ impl<T> flow_relative::Rect<T> {
         // Will be needed for other writing modes
         // FIXME: what if the containing block has a different mode?
         // https://drafts.csswg.org/css-writing-modes/#orthogonal-flows
-        _containing_block: &physical::Rect<T>,
-    ) -> physical::Rect<T>
+        _containing_block: &PhysicalRect<T>,
+    ) -> PhysicalRect<T>
     where
-        T: Clone,
+        T: Clone + Copy,
     {
         // Top-left corner
         let (tl_x, tl_y) = match mode.start_start_physical_corner() {
             PhysicalCorner::TopLeft => (&self.start_corner.inline, &self.start_corner.block),
             _ => unimplemented!(),
         };
-        physical::Rect {
-            top_left: physical::Vec2 {
-                x: tl_x.clone(),
-                y: tl_y.clone(),
-            },
-            size: self.size.size_to_physical(mode),
-        }
-    }
-}
-
-impl<T> physical::Rect<T> {
-    pub fn translate(&self, by: &physical::Vec2<T>) -> Self
-    where
-        T: Add<Output = T> + Copy,
-    {
-        physical::Rect {
-            top_left: &self.top_left + by,
-            size: self.size.clone(),
-        }
+        PhysicalRect::new(
+            PhysicalPoint::new(tl_x.clone(), tl_y.clone()),
+            self.size.to_physical(mode),
+        )
     }
 }
 
-impl physical::Rect<Length> {
-    pub fn axis_aligned_bounding_box(&self, other: &Self) -> Self {
-        let top_left = physical::Vec2 {
-            x: self.top_left.x.min(other.top_left.x),
-            y: self.top_left.y.min(other.top_left.y),
-        };
-
-        let bottom_corner_x = (self.top_left.x + self.size.x).max(other.top_left.x + other.size.x);
-        let bottom_corner_y = (self.top_left.y + self.size.y).max(other.top_left.y + other.size.y);
-        let size = physical::Vec2 {
-            x: bottom_corner_x - top_left.x,
-            y: bottom_corner_y - top_left.y,
-        };
-
-        Self { top_left, size }
-    }
+pub trait ToWebRender {
```

Comment:
Why not use the standard library `From` or `Into` trait?

Reply:
The issue here is that both of the data structures are defined in separate crates. I believe that Into requires at least one type to be defined in the current crate. Perhaps I'm wrong though...